pkg/scraping/client.go

```diff
@@ -54,3 +58,50 @@ func (c *nodeClient) stateHash(ctx context.Context, height int) (*proto.StateHas
 	}
 	return sh, nil
 }
+
+func (c *nodeClient) baseTarget(height int) (int64, error) {
+	url, err := joinUrl(c.cl.GetOptions().BaseUrl, fmt.Sprintf("/blocks/headers/at/%d", height))
```

Comment:
Consider using client function `HeadersAt` pkg/client/blocks.go:109.

Reply:
Completely agree with this issue